tensor_comprehensions/pybinds/tclib.cc

```diff
@@ -447,6 +473,11 @@ PYBIND11_MODULE(tclib, m) {
     return res;
   });
 
+  // Get GPU maximum shared memory
+  m.def("max_shared_memory", []() {
```

Comment:
This is confusing because `max_shared_memory` is a mapping options and here you want to expose properties of the GPUs. I would call it `shared_memory_size`.
Also please split the addition of this function in a separate commit.